lib/ParseSIL/ParseSIL.cpp

```diff
@@ -937,21 +939,44 @@ static bool parseSymbolicValue(SymbolicValue &value, SILParser &SP,
         return true;
 
       // Parse floating point value.
+      // Handle hexadecimal case.
+      if (P.Tok.is(tok::integer_literal)) {
```

Comment:
Need to check that the integer literal is a hex string (e.g. starts with `0x`).

Reply:
Done. Made other fixes in https://github.com/apple/swift/pull/17077/commits/6ab561d71284a8845eda1090959da586a247fa99 as well.